01_classify_stratified/run_cancer_type_classification.py

```diff
@@ -58,38 +80,55 @@ def process_args():
     else:
         not_in_tcga = set(args.cancer_types) - set(tcga_cancer_types)
         if len(not_in_tcga) > 0:
-            p.error('some cancer types not present in TCGA: {}'.format(
-                ' '.join(not_in_tcga)))
+            parser.error('some cancer types not present in TCGA: {}'.format(
+                         ' '.join(not_in_tcga)))
 
-    return args, sample_info_df
+    # split args into defined argument groups, since we'll use them differently
+    arg_groups = du.split_argument_groups(args, parser)
+    io_args, model_options = arg_groups['io'], arg_groups['model_options']
+
+    # add some additional hyperparameters/ranges from config file to model options
+    model_options.alphas = cfg.alphas
+    model_options.l1_ratios = cfg.l1_ratios
+    model_options.standardize_data_types = cfg.standardize_data_types
+
+    return io_args, model_options, sample_info_df
 
 
 if __name__ == '__main__':
 
     # process command line arguments
-    args, sample_info_df = process_args()
+    io_args, model_options, sample_info_df = process_args()
 
     # create results dir if it doesn't exist
-    args.results_dir.mkdir(parents=True, exist_ok=True)
+    io_args.results_dir.mkdir(parents=True, exist_ok=True)
+    # create subdir for experiment if it doesn't exist
+    experiment_dir = Path(io_args.results_dir, 'cancer_type').resolve()
+    experiment_dir.mkdir(parents=True, exist_ok=True)
```

Comment:
> Will `io_args.results_dir` always contain a `Path` object? Not sure how `argparse` handles this. I thought you might get a `str` instance here and thus `mkdir` would fail.

The argparse input should be converted from a `str` to a `Path` object in the `process_args` function (see line 69), so I think this should work fine as is. Good to think about it though!

> Also, since you are using `parents=True`, you can safely remove line 104 I think (and avoid the potential issue I mention above).

Good catch!